opacclient/libopac/src/main/java/de/geeksfactory/opacclient/apis/Adis.java

```diff
@@ -1614,11 +1639,22 @@ protected Document handleLoginForm(Document doc, Account account)
                                .select("option");
         }
         for (Element opt : searchoptions) {
-            TextSearchField field = new TextSearchField();
-            field.setId(opt.attr("value"));
-            field.setDisplayName(opt.text());
-            field.setHint("");
-            fields.add(field);
+            // Damit doppelte Optionen nicht mehrfach auftauchen
+            // (bei Stadtbücherei Stuttgart der Fall)
+            boolean found = false;
+            for (SearchField f : fields) {
+                if (f.getDisplayName().equals(opt.text())) {
+                    found = true;
+                    break;
+                }
+            }
```

Comment:
I don't know if it is worth it just keeping a HashSet of texts here? This seems pretty inefficient it needlessly goes with O(n²). I'm unsure if it matters with the number of fields we usually have.

Also, shouldn't we check for equal IDs instead? Or do they in fact have different ones?

Reply:
Makes sense, done in c6b62dea (this implementation was just copied from @JustyAnOther 's PR, I didn't think about if there's an easier way.